docs/src/index.md

```diff
@@ -268,6 +268,178 @@ df2 = @eachrow df begin
 end
 ```
 
+## Row-wise transformations with `@byrow`
+
+DataFrames provides the function-wrapper `ByRow`. `ByRow(f)(x, y)`
+is roughly equivalent to `f.(x, y)`, with a few exceptions discussed below. 
+DataFramesMeta allows for users to construct expressions using `ByRow` 
+function wrapper with the syntax `@byrow`. 
+
+```julia
+@transform(df, y = @byrow :x == 1 ? "true" : "false)
+```
+
+becomes
+
+```
+transform(df, :x => ByRow(x -> x == 1 ? "true", "false") => :y)
+```
+
+!!! note 
+    Unlike `@.`, `@byrow` is not a "real" macro and cannot be used outside of 
+    DataFramesMeta macros. However it's behavior within DataFramesMeta 
+    macros should be indistinguishable from externally defined macros. 
+
+### Comparison with `@eachrow`
+
+In previous versions of DataFramesMeta, `@eachrow` was named `@byrow`. 
+This version of `@byrow` is deprecated, but the syntax can be used
+to for similar, but not identical, behavior.
+
+The syntax 
+
+```julia
+@eachrow df begin 
+    :a * :b
+end 
+```
+
+is similar to 
+
+```julia
+begin
+    function tempfun(a, b)
+        for i in eachindex(a)
+            a[i] * b[i]
+        end
+    end
+    tempfun(df.a, df.b)
+    df
+end
+```
+
+The function `*` is applied by-row. But the result of those operations
+is not stored in a new vector. Additionally, `@eachrow` and `@eachrow!`
+return data frames. 
+
+By contrast,
+
+```julia
+@with df @byrow begin 
```

Comment:
this is confusing. I understand you refer to the deprecated `@byrow`? If this is the case then I would skip it - no need to document (or even mention deprecated functionality except for writing that it is deprecated).

If you refer to new `@byrow` then it is strange, as earlier you have written that `@byrow` is restricted to `@transform`.

Reply:
Looking at your examples below it seems `@byrow` is accepted not only in `@transform` - I would suggest explicitly listing places where it is allowed.

Reply:
Added.